### PR TITLE
Delete all jobs API

### PR DESCRIPTION
**Pull request by zerodivisi0n**
This PR adds the ability to delete all jobs at once, as proposed in #147.

I decided to call method `DELETE /job/all`, although it is not quite suitable for REST. But I think it fits better than `DELETE /job`. 

I used the easiest way to implement the deletion - simply call `Delete` on each job in a loop. Another way to do it is to add methods `DeleteAll` to `Db` and `Cache`, but it extends interface and requires more work to be done.

Also, I added this method to the client.

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -186,6 +186,18 @@ func HandleJobRequest(cache job.JobCache, db job.JobDB) func(w http.ResponseWrit
 	}
 }
 
+// HandleDeleteAllJobs is the handler for deleting all jobs
+// DELETE /api/v1/job/all
+func HandleDeleteAllJobs(cache job.JobCache, db job.JobDB) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if err := job.DeleteAll(cache, db); err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
```
Comment (elimisteve) [inline api/api.go]: @ajvb What should be done with errors in situations like this? Should be fine to send to the user as JSON, yes? E.g., `{"error": "..."}`

Comment (ajvb) [reply api/api.go]: Yeah absolutely, will want to use something similar to `http.Error` - https://github.com/ajvb/kala/pull/160/files#diff-604f4192deea5adf9c159e5f88f64630R130 - as that is what we are currently using.

I'm going to make an issue for the fact that we should actually send back json

Comment (zerodivisi0n) [reply api/api.go]: I did it like it was done for simple deletion endpoint at https://github.com/ajvb/kala/pull/160/files#diff-604f4192deea5adf9c159e5f88f64630R179
But I can fix it after merging #165.

Comment (ajvb) [reply api/api.go]: :+1: Makes sense. #165 is now merged in

Comment (zerodivisi0n) [reply api/api.go]: Done!

**Comment by ajvb** [comment]
Awesome, thank you so much @zerodivisi0n!

**Pull request merged**